privacyidea/lib/tokens/radiustoken.py

```diff
@@ -143,15 +144,193 @@ def update(self, param):
         self.add_tokeninfo("radius.user", val)
         self.add_tokeninfo("tokenkind", TOKENKIND.VIRTUAL)
 
+        log.debug('***[radius token: update]')
+
+    def is_challenge_request(self, passw, user=None, options=None):
+        """
+        This method checks, if this is a request, that triggers a challenge.
+        It depends on the way, the pin is checked - either locally or remote
+
+        :param passw: password, which might be pin or pin+otp
+        :type passw: string
+        :param user: The user from the authentication request
+        :type user: User object
+        :param options: dictionary of additional request parameters
+        :type options: dict
+
+        :return: true or false
+        """
+        otp_counter = -1
+        otpval = passw
+        state = options.get('radius_state')
+
+        log.debug("***[radius token: is challenge request]")
+
+        # should we check the pin locally?
+        if self.check_pin_local:
+            (_res, pin, otpval) = self.split_pin_pass(passw, user,
+                                                      options=options)
+            if not TokenClass.check_pin(self, pin):
```

Comment:
why not do ``self.check_pin(pin)``?

Reply:
I will change this.  You'll have to excuse me when my Java background comes through.